pkg/controllers/deployment/deployment_controller.go

```diff
@@ -362,3 +352,33 @@ func (c *deploymentController) ensureBootstrappedOAuthClients(ctx context.Contex
 
 	return nil
 }
+
+// updateOperatorDeploymentInfo updates the operator's Status .ReadyReplicas, .Generation and the
+// .Generetions field with data about the oauth-server deployment
+func (c *deploymentController) updateOperatorDeploymentInfo(
+	ctx context.Context,
+	syncContext factory.SyncContext,
+	operatorConfig *operatorv1.Authentication,
+	deployment *appsv1.Deployment,
+) error {
+	operatorStatusOutdated := operatorConfig.Status.ObservedGeneration != operatorConfig.Generation || operatorConfig.Status.ReadyReplicas != deployment.Status.UpdatedReplicas
```

Comment:
why we are not using [the generic workload controller](https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/controller/workload/workload.go) for this deployment? 

The generic controller is used to deploy the openshift-apiserver and the new oauth-apiserver